server/utils/courseParticipationUtils.ts

```diff
@@ -83,79 +83,95 @@ export default class CourseParticipationUtils {
   private static summaryListRows(
     courseParticipationWithName: CourseParticipationWithName,
   ): Array<GovukFrontendSummaryListRowWithValue> {
-    const summaryListRows: Array<GovukFrontendSummaryListRowWithValue> = []
+    return [
+      {
+        key: { text: 'Programme name' },
+        value: { text: courseParticipationWithName.name },
+      },
+      CourseParticipationUtils.settingSummaryListRow(courseParticipationWithName),
+      CourseParticipationUtils.outcomeSummaryListRow(courseParticipationWithName),
+      CourseParticipationUtils.outcomeAdditionalDetailSummaryListRow(courseParticipationWithName),
+      CourseParticipationUtils.sourceSummaryListRow(courseParticipationWithName),
+      CourseParticipationUtils.addedBySummaryListRow(courseParticipationWithName),
+    ]
+  }
 
-    summaryListRows.push({
-      key: { text: 'Programme name' },
-      value: { text: courseParticipationWithName.name },
-    })
+  // eslint-disable-next-line @typescript-eslint/member-ordering
+  private static settingSummaryListRow(
+    courseParticipationWithName: CourseParticipationWithName,
+  ): GovukFrontendSummaryListRowWithValue {
+    const valueTextItems: Array<string> = []
 
-    if (courseParticipationWithName.setting) {
-      const valueTextItems: Array<string> = []
+    if (courseParticipationWithName.setting.type) {
+      valueTextItems.push(StringUtils.properCase(courseParticipationWithName.setting.type))
+    }
 
-      if (courseParticipationWithName.setting.type) {
-        valueTextItems.push(StringUtils.properCase(courseParticipationWithName.setting.type))
-      }
+    if (courseParticipationWithName.setting.location) {
+      valueTextItems.push(`${courseParticipationWithName.setting.location}`)
+    }
 
-      if (courseParticipationWithName.setting.location) {
-        valueTextItems.push(`${courseParticipationWithName.setting.location}`)
-      }
+    return {
+      key: { text: 'Setting' },
+      value: { text: valueTextItems.join(', ') || 'Not known' },
+    }
```

Comment:
I wonder if it's worth abstracting this into something like:

```ts
  private static summaryListRow(keyText: GovukFrontendSummaryListRowKey['text'], valueTextItems: Array<string>): GovukFrontendSummaryListRow {
    return {
      key: { text: keyText },
      value: { text: valueTextItems.join(', ') || 'Not known' },
    }
}
```

Then it can be reused in all of them including the `addedBy` one, and the programme name row could also use it